site/src/pages/WorkspaceSchedulePage/WorkspaceSchedulePage.tsx

```diff
@@ -167,9 +116,66 @@ export const WorkspaceSchedulePage: React.FC = () => {
     username && workspaceName && scheduleSend({ type: "GET_WORKSPACE", username, workspaceName })
   }, [username, workspaceName, scheduleSend])
 
+  const getAutoStart = (workspace?: TypesGen.Workspace) =>
+    scheduleToAutoStart(workspace?.autostart_schedule)
+  const getAutoStop = (workspace?: TypesGen.Workspace) => ttlMsToAutoStop(workspace?.ttl_ms)
+
+  const [autoStart, setAutoStart] = useState(getAutoStart(workspace))
+  const [autoStop, setAutoStop] = useState(getAutoStop(workspace))
+
+  useEffect(() => {
+    setAutoStart(getAutoStart(workspace))
+    setAutoStop(getAutoStop(workspace))
+  }, [workspace])
+
+  const onToggleAutoStart = () => {
+    if (autoStart.enabled) {
+      setAutoStart({
+        enabled: false,
+        schedule: emptySchedule,
+      })
+    } else {
+      if (workspace?.autostart_schedule) {
+        // repopulate saved schedule
+        setAutoStart({
+          enabled: true,
+          schedule: getAutoStart(workspace).schedule,
+        })
+      } else {
+        // populate with defaults
+        setAutoStart({
+          enabled: true,
+          schedule: defaultSchedule(),
+        })
+      }
+    }
+  }
+
+  const onToggleAutoStop = () => {
+    if (autoStop.enabled) {
+      setAutoStop({
+        enabled: false,
+        ttl: emptyTTL,
+      })
+    } else {
+      if (workspace?.ttl_ms) {
+        // repopulate saved ttl
+        setAutoStop({
+          enabled: true,
+          ttl: getAutoStop(workspace).ttl,
+        })
+      } else {
+        // set default
+        setAutoStop({
+          enabled: true,
+          ttl: defaultTTL,
+        })
+      }
+    }
+  }
```

Comment:
This file is getting kind of big - could we move some of this stuff into `schedule.ts` and then just call those utils in the jsx?

Reply:
What if I move out the other util functions? These are a little more convenient here so they don't need an argument.

Reply:
Sure, sounds good to me!